lib/cnet/ipv4/ip4_input.c

```diff
@@ -106,10 +106,16 @@ ip4_input_node_process(struct cne_graph *graph, struct cne_node *node, void **ob
         dip[0] = dip[1] = dip[2] = dip[3] = 0;
 
         /* Extract DIP from mbufs plus validate the IP header checksum. */
-        ip4[0] = pktmbuf_adjust(mbuf0, struct cne_ipv4_hdr *, mbuf0->l2_len);
-        ip4[1] = pktmbuf_adjust(mbuf1, struct cne_ipv4_hdr *, mbuf1->l2_len);
-        ip4[2] = pktmbuf_adjust(mbuf2, struct cne_ipv4_hdr *, mbuf2->l2_len);
-        ip4[3] = pktmbuf_adjust(mbuf3, struct cne_ipv4_hdr *, mbuf3->l2_len);
+        ip4[0] = pktmbuf_mtod(mbuf0, struct cne_ipv4_hdr *);
+        ip4[1] = pktmbuf_mtod(mbuf1, struct cne_ipv4_hdr *);
+        ip4[2] = pktmbuf_mtod(mbuf2, struct cne_ipv4_hdr *);
+        ip4[3] = pktmbuf_mtod(mbuf3, struct cne_ipv4_hdr *);
+
+        /* Adjust the data length for an IPv4 packet to the size given in the header */
+        pktmbuf_data_len(mbuf0) = be16toh(ip4[0]->total_length);
```

Comment:
Should you trust the network packet here? I think we should at least verify that ip4.total_length is <= the maximum length available in the mbuf. For example, if the length were 0xFFFF, that would be bad.

Reply:
Added test for length to be valid.